src/modules/authentication/electron.oidc.authentication.service.ts

```diff
@@ -1,135 +1,98 @@
-import {Router} from 'aurelia-router';
+
+import {IIdentity} from '@essential-projects/iam_contracts';
 
 import {AuthenticationStateEvent,
         IAuthenticationService,
-        IIdentity,
-        ISolutionEntry,
+        ILoginResult,
         ISolutionService,
         ITokenObject,
+        IUserIdentity,
         NotificationType} from '../../contracts/index';
-import environment from '../../environment';
 import {NotificationService} from '../notification/notification.service';
 
 const UNAUTHORIZED_STATUS_CODE: number = 401;
 const IDENTITY_SERVER_AVAILABLE_SUCCESS_STATUS_CODE: number = 200;
 
-@inject(EventAggregator, 'NotificationService', Router, 'SolutionService')
+@inject(EventAggregator, 'NotificationService')
 export class ElectronOidcAuthenticationService implements IAuthenticationService {
 
   private _eventAggregator: EventAggregator;
   private _notificationService: NotificationService;
-  private _router: Router;
-  private _tokenObject: ITokenObject;
-  private _solutionService: ISolutionService;
 
   constructor(eventAggregator: EventAggregator,
-              notificationService: NotificationService,
-              router: Router,
-              solutionService: ISolutionService) {
+              notificationService: NotificationService) {
     this._eventAggregator = eventAggregator;
     this._notificationService = notificationService;
-    this._router = router;
-    this._solutionService = solutionService;
-
-    this._getPersistedTokenObject();
-
-    const tokenObjectIsNotUndefined: boolean = this._tokenObject !== null;
-    if (tokenObjectIsNotUndefined) {
-      this.checkUserInfo();
-    }
   }
 
-  public isLoggedIn(): boolean {
-    const userIsExisting: boolean = this._tokenObject !== undefined;
+  public async isLoggedIn(authority: string): Promise<boolean> {
 
-    return userIsExisting;
+    return false;
   }
 
-  public async login(): Promise<void> {
-
-    const identityServerIsNotReachable: boolean = await !this._isIdentityServerReachable();
-    if (identityServerIsNotReachable) {
-      this._notificationService.showNotification(NotificationType.ERROR, 'IdentityServer is offline');
-      return;
-    }
+  public async login(authority: string): Promise<ILoginResult> {
 
-    const ipcRenderer: any = (window as any).nodeRequire('electron').ipcRenderer;
+    const loginResultPromise: Promise<ILoginResult> = new Promise(async(resolve: Function, reject: Function): Promise<void> => {
 
-    ipcRenderer.on('oidc-login-reply', async(event: any, tokenObject: ITokenObject) => {
-      this._tokenObject = tokenObject;
-      const identity: IIdentity = await this.getIdentity();
-      this._eventAggregator.publish(AuthenticationStateEvent.LOGIN, identity);
+      const identityServerIsNotReachable: boolean = await !this._isAuthorityReachable(authority);
+      if (identityServerIsNotReachable) {
+        this._notificationService.showNotification(NotificationType.ERROR, 'IdentityServer is offline');
```

Comment:
Would prefer here something like that

```suggestion
        this._notificationService.showNotification(NotificationType.ERROR, 'Can not connect to the IdentityServer.');
```

Reply:
`Can not connect to the IdentityServer.` is not as helpful for the user, because there is no reason why the BPMN-Studio can not connect to the IdentityServer.

Reply:
@SteffenKn Yes but the other message isn't helpful too. If the request fails it doesn't mean for sure that the identity server is offline

Reply:
I think @alexanderkasten is right here.